bundles/binding/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/SagerCasterBinding.java

```diff
@@ -0,0 +1,245 @@
+import org.joda.time.base.AbstractInstant;
+import org.openhab.binding.sagercaster.SagerCasterBindingProvider;
+import org.openhab.core.binding.AbstractBinding;
+import org.openhab.core.items.GenericItem;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.items.ItemRegistryChangeListener;
+import org.openhab.core.items.StateChangeListener;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.persistence.HistoricItem;
+import org.openhab.core.persistence.extensions.PersistenceExtensions;
+import org.openhab.core.types.State;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+	
+
+/**
+ * Sager Weather Caster Binding implementation
+ * 
+ * @author Gaël L'hopital
+ * @since 1.7.0
+ */
+public class SagerCasterBinding extends AbstractBinding<SagerCasterBindingProvider> implements ItemRegistryChangeListener, StateChangeListener {
+
+	private static final Logger logger = LoggerFactory.getLogger(SagerCasterBinding.class);
+	private static SagerWeatherCaster sagerWeatherCaster = null;
+	
+	private ItemRegistry itemRegistry;
+	
+	static final String WINDBEARING = "windbearing";
+	static final String SEALEVELPRESSURE = "sealevelpressure";
+	static final String CLOUDLEVEL = "cloudlevel";
+	static final String RAINING = "raining";
+	static final String WINDSPEED = "windspeed";
+	static final String LATITUDE = "latitude";
+	
+	Map<String, String> itemNames = new HashMap<String, String>();	// Link between configuration name and item name
+
+	public SagerCasterBinding() {
+		sagerWeatherCaster = new SagerWeatherCaster();
+	}
+			
+	/**
+	 * Called by the SCR to activate the component with its configuration read from CAS
+	 * 
+	 * @param bundleContext BundleContext of the Bundle that defines this component
+	 * @param configuration Configuration properties for this component obtained from the ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext, final Map<String, Object> configuration) {
+		modified(configuration);
+	}
+	
+	/**
+	 * Called by the SCR when the configuration of a binding has been changed through the ConfigAdmin service.
+	 * @param configuration Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+		itemNames.clear();
+		// the wind bearing item must be persisted via the default persistence service
+		itemNames.put((String) configuration.get(WINDBEARING),WINDBEARING);	
+		// the sea level pressure item must be persisted via the default persistence service
+		itemNames.put((String) configuration.get(SEALEVELPRESSURE),SEALEVELPRESSURE);
+		itemNames.put((String) configuration.get(CLOUDLEVEL),CLOUDLEVEL);
+		itemNames.put((String) configuration.get(RAINING),RAINING);
+		itemNames.put((String) configuration.get(WINDSPEED),WINDSPEED);
```

Comment:
if i got you right, this code wires an item name with the according weather key. Why not simply use the "normal" binding configuration. This gives the user the flexibility to choose their favourite item name. The items have to be creating anyhow.